replica/table.cc

```diff
@@ -864,7 +870,7 @@ compaction_group* table::single_compaction_group_if_available() const noexcept {
 }
 
 compaction_group* table::get_compaction_group(size_t id) const noexcept {
-    return storage_groups()[id]->main_compaction_group().get();
+    return storage_groups().at(id)->main_compaction_group().get();
```

Comment:
maybe more elegantly, it should check for the storage_group existence first, and call on_fatal_internal_error if it doesn't exist - this will produce a better error report than terminating due to exception in noexcept context.